js/util/evented.js

```diff
@@ -95,6 +96,18 @@ var Evented = {
     },
 
     /**
+     * Asyncronously fires an event of the specified type. The event will be
+     * fired within a "setTimeout(..., 0)" callback.
+     *
+     * @param {string} type The type of event to fire.
+     * @param {Object} [data] Data to be passed to any listeners.
+     * @returns {Object} `this`
+     */
+    asyncFire: function(type, data) {
+        window.setTimeout(this.fire.bind(this, type, data), 0);
```

Comment:
@jfirebaugh How do you feel about this use of `setTimeout`? There are some situations where we cannot fire an event synchronously (namely in a constructor).